### PR TITLE
feat(HACBS-2538): collect-data emits paths to generated files

### PR DESCRIPTION
**Pull request by johnbieren**

### CODE REVIEW

File: catalog/task/collect-data/0.3/collect-data.yaml [hunk 1]
```diff
@@ -61,17 +75,24 @@ spec:
           mkdir -p $SUBDIR
         fi
 
+        echo -n "${SUBDIR}" > $(results.resultDirectory.path)
+
         cd $SUBDIR
 
+        echo -n "${SUBDIR}/release.json" > $(results.release.path)
```
Comment (mmalina) [inline catalog/task/collect-data/0.3/collect-data.yaml:82]: This is wrong. This definitely all needs to be relative to the workspace directory. You cannot work with absolute paths among multiple tasks. If just for the fact that workspace names can be different (config vs. data). But even if we always used the same, we should always rely on `$(workspaces.data.path)` or similar inside tasks.

Comment (johnbieren) [reply catalog/task/collect-data/0.3/collect-data.yaml]: Good call, fixed. It does make me question why we want the `resultsDirectory` result though when we already have `params.subdirectory` and they will be the same

Comment (mmalina) [reply catalog/task/collect-data/0.3/collect-data.yaml]: Yeah, I don't think we need that one. I think we just need each of the files.

Comment (davidmogar) [reply catalog/task/collect-data/0.3/collect-data.yaml]: Yep.

Comment (johnbieren) [reply catalog/task/collect-data/0.3/collect-data.yaml]: Removed resultDirectory result

File: catalog/task/collect-data/0.3/README.md [hunk 2]
```diff
@@ -18,5 +25,9 @@ This task also stores the passed resources as json files in a workspace.
 | snapshot             | Namespaced name of the Snapshot                    | No       | -             |
 | subdirectory         | Subdirectory inside the workspace to be used.      | Yes      | -             |
 
+## Changes since 0.2
+  * Added task results for the path to each of the stored resources and one to the subdirectory containing
```
Comment (mmalina) [inline catalog/task/collect-data/0.3/README.md:29]: subdirectory is no longer included in the results

File: catalog/task/collect-data/0.3/README.md [hunk 3]
```diff
@@ -7,6 +7,13 @@ a json file called `extra-data.json` in the workspace.
 
 This task also stores the passed resources as json files in a workspace.
 
+The parameters to this task are lowercase instead of camelCase because they are passed from the operator, and the
+operator passes them as lowercase.
+
+A task result is returned for each resource with the relative path to the stored JSON for it in the workspace. There is
+also a task result for the fbcFragment extracted from the snapshot's first component and one specifying the subdirectory
```
Comment (mmalina) [inline catalog/task/collect-data/0.3/README.md:14]: subdirectory is no longer included in the results

File: catalog/task/collect-data/0.3/collect-data.yaml [hunk 4]
```diff
@@ -63,15 +75,20 @@ spec:
 
         cd $SUBDIR
 
+        echo -n "$(params.subdirectory)/release.json" > $(results.release.path)
```
Comment (mmalina) [inline catalog/task/collect-data/0.3/collect-data.yaml:78]: I wonder if we could streamline this a bit. How about we ditch the `SUBDIR` env var and do this:
```sh
if [ -n "$(params.subdirectory)" ]; then
  mkdir -p $(workspaces.data.path)/$(params.subdirectory)
fi

RELEASE_PATH=$(params.subdirectory)/release.json
echo -n $RELEASE_PATH > $(results.release.path)
get-resource "release" "$RELEASE" > $(workspaces.data.path)/$RELEASE_PATH
```

And similar for the rest.

But only if you find it better like I do :)

File: catalog/task/collect-data/0.3/README.md (hunk 2 above)
Comment (johnbieren) [reply catalog/task/collect-data/0.3/README.md]: fixed, thanks

File: catalog/task/collect-data/0.3/collect-data.yaml (hunk 4 above)
Comment (johnbieren) [reply catalog/task/collect-data/0.3/collect-data.yaml]: I don't have much of an opinion on it, so I changed it to like you said

File: catalog/task/collect-data/0.3/collect-data.yaml [hunk 5]
```diff
@@ -51,30 +63,36 @@ spec:
...
+        echo -n $RELEASESTRATEGY_PATH > $(results.releaseStrategy.path)
+        get-resource "releasestrategy" "${RELEASE_STRATEGY}" > $(workspaces.data.path)/$RELEASESTRATEGY_PATH
 
-        get-resource "snapshot" "${SNAPSHOT}" "{.spec}" > "snapshot_spec.json"
+        SNAPSHOTSPEC_PATH=$(params.subdirectory)/snapshot_spec.json
+        echo -n $SNAPSHOTSPEC_PATH > $(results.snapshotSpec.path)
+        get-resource "snapshot" "${SNAPSHOT}" "{.spec}" > $(workspaces.data.path)/$SNAPSHOTSPEC_PATH
 
-        cat "snapshot_spec.json" | jq -cr '.components[0].containerImage' | tr -d "\n" \
+        cat $SNAPSHOTSPEC_PATH | jq -cr '.components[0].containerImage' | tr -d "\n" \
```
Comment (mmalina) [inline catalog/task/collect-data/0.3/collect-data.yaml:95]: I believe you need `$(workspaces.data.path)/$SNAPSHOTSPEC_PATH` here.

File: catalog/task/collect-data/0.3/collect-data.yaml (hunk 4 above)
Comment (mmalina) [reply catalog/task/collect-data/0.3/collect-data.yaml]: ok, cool :)

File: catalog/task/collect-data/0.3/collect-data.yaml (hunk 5 above)
Comment (johnbieren) [reply catalog/task/collect-data/0.3/collect-data.yaml]: right, good catch

**Review by mmalina — APPROVED** [review]

**Pull request merged**